### PR TITLE
[bug] Round Decimal to 9 places to support DynamoDB constraints

### PR DESCRIPTION
**Pull request by domsammut**
**Proposed changes**:
No rounding is currently performed prior to saving to DynamoDB tracker on Decimal values. This can result in saving errors if a value is greater than [38 characters](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Limits.html#limits-data-types).

Rounded to 9 decimal places as per @akelad's recommendation 🙂 

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [x] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: rasa/core/utils.py [hunk 1]
```diff
@@ -477,7 +477,7 @@ def replace_floats_with_decimals(obj: Union[List, Dict]) -> Any:
             obj[j] = replace_floats_with_decimals(obj[j])
         return obj
     elif isinstance(obj, float):
-        return Decimal(obj)
+        return round(Decimal(obj), 9)
```
Comment (wochinge) [inline rasa/core/utils.py]: 7 should even be enough (I just checked how many decimals `time.time()` returns

File: rasa/core/utils.py [hunk 2]
```diff
@@ -465,7 +465,7 @@ def replace_floats_with_decimals(obj: Union[List, Dict]) -> Any:
     Args:
         obj: A `List` or `Dict` object.
 
-    Returns: An object with all matching values and `float` type replaced by `Decimal`.
+    Returns: An object with all matching values and `float` type replaced by `Decimal` rounded to 9 decimal places.
```
Comment (wochinge) [inline rasa/core/utils.py]:
```suggestion
    Returns: An object with all matching values and `float` type replaced by `Decimal` rounded to 9 decimal places.
```
```suggestion
    Returns: An object with all matching values and `float` types replaced by `Decimal`s rounded to 9 decimal places.
```

File: tests/core/test_utils.py [hunk 3]
```diff
@@ -98,6 +98,7 @@ def test_float_conversion_to_decimal():
     d = {
         "int": -1,
         "float": 2.1,
+        "float_round": 0.918040672051180450807805755175650119781494140625,
```
Comment (wochinge) [inline tests/core/test_utils.py]: could you please to the test with an output of `time.time()`. I think that's the more realistic setting for the tracker store.

File: rasa/core/utils.py (hunk 1 above)
Comment (wochinge) [inline rasa/core/utils.py]: can we make the number of decimals a parameter of the function? `9` could be the default value

Comment (domsammut) [reply rasa/core/utils.py]: This function will also consume confidence scores, hence the slightly higher value:

```
{'event': 'user', 'timestamp': Decimal('1579507733.1107571125030517578125'), 'text': 'hi', 'parse_data': {'intent': {'name': 'greet', 'confidence': Decimal('0.918040672051180450807805755175650119781494140625')}
```

File: rasa/core/utils.py [hunk 4]
```diff
@@ -458,14 +458,15 @@ def handler(fut: Future) -> None:
...
+def replace_floats_with_decimals(obj: Union[List, Dict], round_digits: int = 9) -> Any:
     """
     Utility method to recursively walk a dictionary or list converting all `float` to `Decimal` as required by DynamoDb.
 
     Args:
         obj: A `List` or `Dict` object.
+        round_digits: A int value to set the rounding precision of Decimal values.
 
-    Returns: An object with all matching values and `float` type replaced by `Decimal`.
+    Returns: An object with all matching values and `float` types replaced by `Decimal`s rounded to 9 decimal places.
```
Comment (wochinge) [inline rasa/core/utils.py]:
```suggestion
    Returns: An object with all matching values and `float` types replaced by `Decimal`s rounded to `round_digits` decimal places.
```

File: rasa/core/utils.py [hunk 5]
```diff
@@ -458,14 +458,15 @@ def handler(fut: Future) -> None:
     return handler
 
 
-def replace_floats_with_decimals(obj: Union[List, Dict]) -> Any:
+def replace_floats_with_decimals(obj: Union[List, Dict], round_digits: int = 9) -> Any:
```
Comment (wochinge) [inline rasa/core/utils.py]: optional: What do you think about this name?
```suggestion
def replace_floats_with_decimals(obj: Union[List, Dict], decimal_precision: int = 9) -> Any:
```

Comment (domsammut) [reply rasa/core/utils.py]: Haha - originally had that. I opted for digits as the round function uses [ndigits](https://docs.python.org/3/library/functions.html#round) as the second optional parameter descriptor.

Additionally, I didn't want to confuse with [getcontext().prec](https://docs.python.org/2/library/decimal.html).

Happy to change though 🙂

Comment (wochinge) [reply rasa/core/utils.py]: makes sense, good for me then!

File: rasa/core/utils.py (hunk 4 above)
Comment (wochinge) [reply rasa/core/utils.py]: have you seen this one @domsammut ?

**Pull request merged**